src/lamplib/src/genny/tasks/preprocess.py

```diff
@@ -397,14 +432,19 @@ def _parse_actors(self, actors):
     def _parse_clients(self, clients):
         clients_dict = self._recursive_parse(clients)
         for _, client in clients_dict.items():
-            client.setdefault("URI", self._default_uri)
+            if self._mongostream_uri and client.get("Type", None) == ClientType.MONGOSTREAM:
```

Comment:
client.get returns "None" as a default by default so you don't need the second parameter here.